### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by osterman**
## what
* Security group with ingress/egress
* HTTP/HTTPS listeners
* Default target group

## why
* Primary driver is to provide a standard ALB for ECS services

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+  default_action {
+    target_group_arn = "${aws_lb_target_group.default.arn}"
+    type             = "forward"
+  }
+}
+
+resource "aws_lb_listener" "https" {
+  count             = "${var.certificate_arn == "" ? 0 : 1}"
+  load_balancer_arn = "${aws_lb.default.id}"
```
Comment (aknysh) [inline main.tf]: "${aws_lb.default.arn}" ?

File: outputs.tf [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+output "default_target_group_arn" {
+  value = "${aws_lb_target_group.default.arn}"
+}
+
+output "http_listener_arn" {
+  value = "${aws_lb_listener.http.arn}"
+}
+
+output "https_listener_arn" {
+  value = "${aws_lb_listener.https.*.arn}"
```
Comment (aknysh) [inline outputs.tf]: this will return a list type.
let's use `spalt+join` pattern:
"${join("", aws_lb_listener.https.*.arn)}"

File: outputs.tf [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+output "listener_arns" {
```
Comment (aknysh) [inline outputs.tf]: why do we need this output if we have `http_listener_arn`?

File: outputs.tf [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  value = ["${aws_lb_listener.http.arn}"]
+}
+
+output "access_logs_bucket" {
```
Comment (aknysh) [inline outputs.tf]: `access_logs_bucket_d`

File: variables.tf [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+variable "namespace" {
+  description = "Namespace, which could be your organization name, e.g. `cp` or `cloudposse`"
+}
+
+variable "stage" {
+  description = "Stage, e.g. `prod`, `staging`, `dev`, or `test`"
+}
+
+variable "enabled" {
```
Comment (aknysh) [inline variables.tf]: this var not used

File: variables.tf [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+}
+
+variable "tags" {
+  type        = "map"
+  default     = {}
+  description = "Additional tags (e.g. `map(`BusinessUnit`,`XYZ`)"
+}
+
+variable "vpc_id" {
+  type = "string"
```
Comment (aknysh) [inline variables.tf]: add description

File: README.md [hunk 7]
```diff
@@ -1,2 +1,145 @@
-# terraform-aws-alb
-Terraform module to provision a standard ALB for HTTP/HTTP traffic
+![Cloud Posse](https://cloudposse.com/logo-300x69.png)
+
+# terraform-aws-alb [![Build Status](https://travis-ci.org/cloudposse/terraform-aws-alb.svg?branch=master)](https://travis-ci.org/cloudposse/terraform-aws-alb)
```
Comment (osterman) [inline README.md]: Add slack url

File: README.md [hunk 8]
```diff
@@ -1,2 +1,144 @@
...
+
+
+## Outputs
+
+| Name                            | Description                                                     |
+|:--------------------------------|:----------------------------------------------------------------|
+| `alb_arn`                       | The ARN of the ALB                                              |
+| `alb_arn_suffix`                | The ARN suffix of the ALB                                       |
+| `alb_dns_name`                  | DNS name of ALB                                                 |
+| `alb_zone_id`                   | The ID of the zone which ALB is provisioned                     |
```
Comment (aknysh) [inline README.md]: The ID of the zone in which ALB is provisioned

File: README.md [hunk 9]
```diff
@@ -1,2 +1,144 @@
...
+| `http_listener_arn`             | The ARN of the HTTP listener                                    |
+| `https_listener_arn`            | The ARN of the HTTPS listener                                   |
+| `listener_arns`                 | A list of all the listener ARNs                                 |
+| `access_logs_bucket_id`         | The S3 bucket ID for access logs                                |
+
+## Help
+
+**Got a question?**
+
+File a GitHub [issue](https://github.com/cloudposse/terraform-aws-alb/issues), send us an [email](mailto:hello@cloudposse.com) or reach out to us on [Gitter](https://gitter.im/cloudposse/).
```
Comment (aknysh) [inline README.md]: I think we are moving to this format:

Review the [docs](https://docs.cloudposse.com/), ask a [question](https://docs.cloudposse.com/discuss-new), file a GitHub [issue](https://github.com/cloudposse/terraform-aws-ssm-iam-role/issues), send us an [email](mailto:hello@cloudposse.com) or reach out to us on [Slack](https://slack.cloudposse.com).

**Pull request merged**